doc/case-studies/timelock/timelock.tla

```diff
@@ -0,0 +1,30 @@
+(*
+ * Typedefs and variables for the Timelock contract
+ *
+ * Modeled from https://github.com/stellar/soroban-examples/blob/v20.0.0/timelock/src/lib.rs
+ *
+ * Andrey Kuprianov, 2024
+ *)
+---- MODULE timelock ----
+
+EXTENDS env, token
+
+(*
+    @typeAlias: timeBoundKind = Str;
```

Comment:
I'd personally introduce an uninterpreted type here, e.g. `"TBK"` or even `"TIME_BOUND_KIND"`, but it's not blocking